common/event_emitter.go

```diff
@@ -76,6 +79,10 @@ const (
 	// Worker
 
 	EventWorkerClose string = "close"
+
+	// Event listener.
+
+	EventListenerDefaultChanBufferSize int64 = 10
```

Comment:
I would prefer not have to work with buffered channels, but the issue is with handlers which not only listen and handle incoming events, they can also `emit` events too. Without the buffered channel this would deadlock 😭

Reply:
Why it is 10?

Reply:
Why is it `10`?

Reply:
The `10` was a just a random number. I was thinking about this, and actually, it might be better to look at trying to prevent the handler(s) from emitting and receiving event(s), so that we don't need the buffered channels (and leave it as unbuffered). IMO, it doesn't feel right for a handler to be doing these two jobs, and should concentrate on handling the events and not emitting new ones, but I need to look into this a bit more (which could take some time to make sense of).